Create transit line addition and deletion project cards

yueshuaing opened the pull request:
Add infrastructure in Lasso to:

1. [x] Compare two transit.lin file and write out New Transit Route project card that comply with the schema defined here: https://github.com/network-wrangler/projectcard/issues/13
2. [x]  Compare two transit.lin file and write out Delete Transit Service project card

yueshuaing committed:
add new transit project

yueshuaing committed:
create new transit project card by comparing cube line files

i-am-sijia commented on lasso/transit.py:
```diff
@@ -581,22 +655,20 @@ def unpack_route_name(line_name: str):
+        direction_id = parts[1]
 
-        return route_id, time_period, agency_id, direction_id
+        return route_id, direction_id
```
`unpack_route_name()` is also called in `add_additional_time_periods()`, which expects four return objects. Does this change break that function?

i-am-sijia commented on lasso/transit.py:
```diff
@@ -523,6 +537,66 @@ def create_add_route_card_dict(self, line: str):
+                "agency_raw_name":agency_raw_name,
+                "agency_id":agency_id,
+                "trips":[]
+        }
```
Indent looks off.

i-am-sijia commented on lasso/transit.py:
```diff
@@ -523,6 +537,66 @@ def create_add_route_card_dict(self, line: str):
+        }
+
+        for _, row in self.shapes[line].iterrows():
+            if row['stop']:
```
This does not handle the alight, board, and travel time to next node, correct? Perhaps add TODO in the comments.

i-am-sijia reviewed (changes requested):
Thank you, @yueshuaing. Please address my comment. We will merge it to move on with Met Council.

Similar to my Network Wrangler comment, we should add a unit test for this PR.

yueshuaing committed:
addressed the comments in PR#158

added new function to grab route id and direction id from line name, added some comments

yueshuaing committed:
fixed bugs in adding new transit routes

yueshuaing committed:
transit line deletion

yueshuaing committed:
updated indentation

i-am-sijia commented on lasso/transit.py:
```diff
@@ -298,18 +298,33 @@ def evaluate_differences(self, base_transit):
+                route_properties['trips'] = [trip_properties]
+                add_card_dict["routes"].append(route_properties)
+
+        if len(added_routes)>0:
```
What is the reason for adding this filter? What is the case for not appending `add_card_dict` when `lines_to_add` is not empty? Please add comments to this filter.

i-am-sijia reviewed (approved):
I have one comment / question.

yueshuaing committed:
add comments
